File: KKProject/Vender/KKBaseUIKit/KKView/KKView.m
//
//  KKView.m
//  KKForeignTeacher
//
//  Created by 尤彬 on 2018/10/25.
//  Copyright © 2018 youbin. All rights reserved.
//

#import "KKView.h"
#import "AppDelegate.h"

@implementation KKView



-(instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
      [self commitInit];
    }
    return self;
}

- (instancetype)initWithViewModel:(id<KKViewModelProtocol>)viewModel{
    return  [super init];
}

- (void) commitInit{

    [self kk_setupView];
    [self kk_setupConfig];
    [self kk_bindViewModel];
}

-(void)kk_bindViewModel{}

- (void)kk_setupView{}

- (void)kk_setupConfig{}

-(void)kk_addReturnKeyBoard{
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc]
                                   initWithTarget:self
                                   action:@selector(tapAction)];
    tap.numberOfTapsRequired    = 1;
    tap.numberOfTouchesRequired = 1;
    [self addGestureRecognizer:tap];
}


- (void) tapAction{
    AppDelegate *app = (AppDelegate *)UIApplication.sharedApplication.delegate;
    [app.window endEditing:true];
}

-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{}
-(void)touchesEnded:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: KKProject_Swift/Home/KKView/KKView.swift
//
//  KKView.swift
//  KKProject_Swift
//
//  Created by youbin on 2020/1/25.
//  Copyright © 2020 Macbook Pro 15.4 . All rights reserved.
//

import UIKit

class KKView: UIView, KKViewProtocol{
    
    var viewModel : KKViewModelProtocol?
    
    init(viewmodel : KKViewModelProtocol) {
        super.init(frame: CGRect.zero)
        viewModel = viewmodel
        kk_setupView()
        kk_bindViewModel()
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func kk_setupView() {
        
    }
    
    func kk_bindViewModel() {
        
    }

    /*
     // Only ov@objc e@objc rride draw() if you perform custom drawing.
     // An empty implementation adversely affects performance during animation.
     override func draw(_ rect: CGRect) {
     // Drawing code
     }
     */
    
}
